### PR TITLE
API cleanup and uploadManifest tests

### PR DESCRIPTION
**Pull request by lmolkova**

### CODE REVIEW

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java [hunk 1]
```diff
@@ -221,28 +214,27 @@ public Mono<UploadBlobResult> uploadBlob(BinaryData data) {
...
-            .flatMap(completeUploadResponse -> {
-                Response<UploadBlobResult> res = new ResponseBase<ContainerRegistryBlobsCompleteUploadHeaders, UploadBlobResult>(completeUploadResponse.getRequest(),
-                    completeUploadResponse.getStatusCode(),
-                    completeUploadResponse.getHeaders(),
-                    new UploadBlobResult(completeUploadResponse.getDeserializedHeaders().getDockerContentDigest()),
-                    completeUploadResponse.getDeserializedHeaders());
-
-                return Mono.just(res);
-            }).onErrorMap(UtilsImpl::mapException);
+        String digest = computeDigest(data.toByteBuffer());
```
Comment (alzimmermsft) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java:225]: Won't this break if the content is larger than 2GB

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java [hunk 2]
```diff
@@ -221,28 +214,27 @@ public Mono<UploadBlobResult> uploadBlob(BinaryData data) {
...
+        return blobsImpl.startUploadWithResponseAsync(repositoryName, context)
+                .flatMap(startUploadResponse -> blobsImpl.uploadChunkWithResponseAsync(trimNextLink(startUploadResponse.getDeserializedHeaders().getLocation()), data.toFluxByteBuffer(), data.getLength(), context))
```
Comment (alzimmermsft) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java:227]: Does `uploadChunk` have to upload the entire payload in one request?

Comment (lmolkova) [reply sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java:227]: no, it's the next work item on my list, but hope to finish manifest upload/download first

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java (hunk 1 above)
Comment (lmolkova) [reply sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java:225]: stay tuned for the new PR  on this :)

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifactBase.java [hunk 3]
```diff
@@ -1,101 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-// Code generated by Microsoft (R) AutoRest Code Generator.
```
Comment (samvaity) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifactBase.java:3]: Was this an Impl file and we don't need it anymore? 
I see we moved the content to respective sync/async clients. 
Just wanted to know if this was ever a autorest generated file.

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/models/UploadBlobResult.java [hunk 4]
```diff
@@ -3,18 +3,23 @@
...
+    }
 
     private final String digest;
 
     /**
      * Instantiate the upload blob result.
      * @param digest The digest of the blob that was uploaded.
      */
-    public UploadBlobResult(String digest) {
+    private UploadBlobResult(String digest) {
```
Comment (samvaity) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/models/UploadBlobResult.java:22]: breaking?

Comment (lmolkova) [reply sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/models/UploadBlobResult.java:22]: it's in beta, never been GA-ed

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifactBase.java (hunk 3 above)
Comment (lmolkova) [reply sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifactBase.java:3]: good catch! it wasn't auto-generated, I guess the header was copied :)
I removed the generated note from couple of other files were is was incorrectly set.

**Review by samvaity — APPROVED** [review]

**Pull request merged**